sdk/src/client/secret/ledger_nano.rs

```diff
@@ -309,19 +375,13 @@ impl SecretManage for LedgerSecretManager {
             let (remainder_output, remainder_bip32) = match &prepared_transaction.remainders.as_slice() {
                 // Multiple remainder outputs would require blind signing since at least one would contain a native
                 // token, that's why matching a single one is enough.
-                [remainder] => {
-                    if let Some(chain) = remainder.chain {
-                        (
-                            Some(&remainder.output),
-                            LedgerBIP32Index {
-                                bip32_change: chain.change.harden().into(),
-                                bip32_index: chain.address_index.harden().into(),
-                            },
-                        )
-                    } else {
-                        (None, LedgerBIP32Index::default())
-                    }
-                }
+                [remainder] => (
+                    Some(&remainder.output),
+                    LedgerBIP32Index {
+                        bip32_change: chain.change.harden().into(),
+                        bip32_index: chain.address_index.harden().into(),
```

Comment:
Before this was part of the remainder, but now it's just provided to this function which I think can be wrong if the remainder address isn't an Ed25519 address
So if remainder address != Ed25519, return None for the remainder_bip32 and below only go in this part `if let Some(remainder_output) = remainder_output {` if it's Some

Reply:
Since you understand this better, would you push the change?

Reply:
I think it depends on https://github.com/iotaledger/iota-sdk/pull/1756#discussion_r1484276282, if we bring this back, then we can also do this part different

Reply:
that's fair